### PR TITLE
Add feature flags documentation

### PR DESCRIPTION
**Pull request by mrodriguez90**
#### Board:
https://www.notion.so/rootstrap/Documentar-ac6a53fb2ec64aebb818bb9023c8ad11?pvs=4
---
#### Description:
Feature flags documentation
---
#### Notes:
* -
---
#### Tasks:
- [x] Adds feature flags documentation at docs/feature_flags.md
---
#### Risk:
* no
---
#### Preview:

👁️

### CODE REVIEW

File: docs/feature_flags.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Feature Flags
+
+## Intro
+
+The Rootstrap Rails API Base allows developers to define feature flags, which are essentially boolean variables representing whether a particular feature or piece of code should be enabled or disabled. These flags can be easily controlled without requiring code changes or redeployment.
+Feature flags can be set at different levels of granularity, such as for individual users, user groups, or percentage-based rollouts. This allows for controlled feature releases and gradual rollouts.
+Our feature flag engine is built upon Flipper. To learn more about Flipper, please visit the following website: [Flipper Documentation](https://www.flippercloud.io/docs/introduction)
```
Comment (santib) [inline docs/feature_flags.md:7]:
```suggestion
Our feature flag implementation is based on Flipper. To learn more about Flipper, please visit the following website: [Flipper Documentation](https://www.flippercloud.io/docs/introduction)
```

File: docs/feature_flags.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      expect(Flipper).to be_enabled(:test_feature_flag)
+    end
+ end
+```
+
+No need to reset the feature flag state; our system already resets the Flipper instance before each test. This ensures that all tests start in a fresh state.
+
+## Monitoring and Managing Feature Flags
+
+To monitor and manage feature flags, we recommend using the Flipper UI, which can be accessed through `active_admin` or at this path: `/admin/feature-flags/`
```
Comment (santib) [inline docs/feature_flags.md:57]: I would move this section below `Set Up Feature Flags`

File: docs/feature_flags.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# Feature Flags
```
Comment (santib) [inline docs/feature_flags.md:1]: WDYT about adding a section to explain how FFs enable Continuous Deployment to our dev teams? This is a good reference to add: https://about.gitlab.com/handbook/product-development-flow/feature-flag-lifecycle/#the-benefits-of-feature-flags

**Review by santib — APPROVED** [review]
LGTM 👌 Left some comments

**Review by szcrubyroostrap — APPROVED** [review]

**Pull request merged**